crates/router/src/connector/bambora/transformers.rs

```diff
@@ -0,0 +1,273 @@
+use std::fmt::Debug;
+// use common_utils::{pii};
+// use error_stack::{report, IntoReport};
+// use masking::{Secret};
+use serde::{Deserialize, Serialize};
+use crate::{core::errors,pii::PeekInterface,types::{self,api, storage::enums}};
+
+#[derive(Debug, Serialize, Deserialize)]
+enum PaymentMethod {
+    InputCardDetails,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+struct InputCardDetails {
+    name: String,
+    number: String,
+    expiry_month: String,
+    expiry_year: String,
+    cvd: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Custom {
+    ref1: String,
+    ref2: String,
+    ref3: String,
+    ref4: String,
+    ref5: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Card {
+    card_type: String,
+    last_four: String,
+    card_bin: String,
+    address_match: i64,
+    postal_result: i64,
+    avs_result: String,
+    cvd_result: String,
+    cavv_result: Option<String>,
+    avs: Option<Avs>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Avs {
+    id: String,
+    message: String,
+    processed: bool,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Link {
+    rel: String,
+    href: String,
+    method: String,
+}
+
+//TODO: Fill the struct with respective fields
+#[derive(Debug, Serialize, Deserialize)]
+pub struct BamboraPaymentsRequest {
+    amount: i64,
+    payment_method: String,
+    card: InputCardDetails,
+    order_number: String
+}
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for BamboraPaymentsRequest  {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(_item: &types::PaymentsAuthorizeRouterData) -> Result<Self,Self::Error> {
+        let card_details:InputCardDetails = match _item.request.payment_method_data {
+            api::PaymentMethod::Card(ref ccard) => InputCardDetails{
+                name:ccard.card_holder_name.peek().clone(),
```

Comment:
Don't peek the Secrets, Instead use Secret for the fields in `BamboraPaymentsRequest ` wherever necessary